.github/scripts/npm_retention.py

```diff
@@ -0,0 +1,96 @@
+import os
+import requests
+from datetime import datetime, timedelta
+import json
+import subprocess
+import re
+
+PACKAGE_NAME = "@upstash/vector"
+DAYS_TO_KEEP = 14
+CI_VERSIONS_TO_KEEP = 5
+NPM_TOKEN = os.environ.get("NPM_TOKEN")
+
+
+def run_npm_command(command):
+    try:
+        result = subprocess.run(command, capture_output=True, text=True, check=True)
+        return result.stdout.strip()
+    except subprocess.CalledProcessError as e:
+        print(f"Error running command: {e}")
+        print(e.stderr)
+        return None
+
+
+def get_package_versions():
+    output = run_npm_command(["npm", "view", PACKAGE_NAME, "versions", "--json"])
+    if output:
+        return json.loads(output)
+    return []
+
+
+def parse_ci_version_date(version):
+    match = re.search(r"-(\d{14})$", version)
+    if match:
+        date_str = match.group(1)
+        return datetime.strptime(date_str, "%Y%m%d%H%M%S")
+    return None
+
+
+def is_ci_version(version):
+    return bool(re.search(r"-ci\.", version))
+
+
+def deprecate_package_version(version):
+    result = run_npm_command(
+        [
+            "npm",
+            "deprecate",
+            f"{PACKAGE_NAME}@{version}",
+            "This CI version has been deprecated due to retention policy",
+        ]
+    )
+    if result is not None:
```

Comment:
did you try if this if/else works?

Reply:
no i didn't test the whole script yet, only partially. I'll test only if it makes sense overall